content/kb/tutorials/databases/private-gsheet.md

```diff
@@ -116,36 +113,32 @@ Copy the code below to your Streamlit app and run it.
 # streamlit_app.py
 
 import streamlit as st
-from google.oauth2 import service_account
-from gsheetsdb import connect
+from streamlit_gsheets import GSheetsConnection
 
 # Create a connection object.
-credentials = service_account.Credentials.from_service_account_info(
-    st.secrets["gcp_service_account"],
-    scopes=[
-        "https://www.googleapis.com/auth/spreadsheets",
-    ],
-)
-conn = connect(credentials=credentials)
-
-# Perform SQL query on the Google Sheet.
-# Uses st.cache_data to only rerun when the query changes or after 10 min.
-@st.cache_data(ttl=600)
-def run_query(query):
-    rows = conn.execute(query, headers=1)
-    rows = rows.fetchall()
-    return rows
+conn = st.experimental_connection("gsheets", type=GSheetsConnection)
 
-sheet_url = st.secrets["private_gsheets_url"]
-rows = run_query(f'SELECT * FROM "{sheet_url}"')
+df = conn.read(
+    worksheet="Sheet1",  # The first worksheet is used if not specified.
+    ttl="10m",
+    usecols=[0, 1],
+)
+df.dropna(inplace=True)
```

Comment:
```suggestion
df = conn.read(
    worksheet="Sheet1",  # The first worksheet is used if not specified.
    ttl="10m",
    usecols=[0, 1],
    nrows=3,
)
```

How about using `nrows` rather than inplace operations we don't encourage due to Streamlit's execution model?

Reply:
The funny thing is that null rows are supposed to be ignored by default and the extra operation shouldn't be needed at all. But in executing the code, sometimes it grabs the empty rows and sometimes it doesn't. There may be some specific version involved in switching the behavior, and may even be a bug, but I haven't pinpointed it yet.